### PR TITLE
kernel: Fix unused-parameter warnings

### PR DESCRIPTION
**Pull request by kartben**
Add missing ARG_UNUSED where needed.

### CODE REVIEW

File: kernel/queue.c [hunk 1]
```diff
@@ -88,6 +88,9 @@ static inline void handle_poll_events(struct k_queue *queue, uint32_t state)
 {
 #ifdef CONFIG_POLL
 	z_handle_obj_poll_events(&queue->poll_events, state);
+#else
+	ARG_UNUSED(queue);
+	ARG_UNUSED(state);
```
Comment (andyross) [inline kernel/queue.c:93]: Don't put these inside #if conditions.  They're harmless in all contexts, just put them up at the top of the function where they don't get in the way.

**Review by andyross — COMMENTED** [review]
One nitpick.  But IIRC I remember from some discussion on Discord a while back that we're supposed to be preferring the `__maybe_unused` attribute now?  For sure it's clearer to read.  @stephanosio might know?

File: kernel/queue.c (hunk 1 above)
Comment (kartben) [reply kernel/queue.c:93]: mmm fwiw I just looked at how it seemed to be done in other places in the kernel and e.g. `thread.c` definitely uses the pattern where `ARG_UNUSED` is inside #if conditions, so it should probably be consistent at least throughout a file?

**Comment by kartben** [comment]
> One nitpick. But IIRC I remember from some discussion on Discord a while back that we're supposed to be preferring the `__maybe_unused` attribute now? For sure it's clearer to read. @stephanosio might know?

Oh, I have no idea, there seems to be just a handful `__maybe_unused` in-tree at the moment, vs 1000s of occurences of ARG_UNUSED, so if the former is now the recommended approach we are probably not doing a good job at promoting it.

**Comment by kartben** [comment]
@andyross could you have another look please?

**Review by cfriedt — APPROVED** [review]
LGTM

**Review by peter-mitsis — APPROVED** [review]
Speculation--feel free to ignore. I don't know of any official preference for using the attributes over ARG_UNUSED(). However, since we do have "__maybe_unused", should we also have "ARG_MAYBE_UNUSED()" for both symmetry and to help reduce #if/#endif clutter.

**Review by andyross — APPROVED** [review]
Not worth arguing about style

**Pull request merged**